packages/core/utils/src/dml/types.ts

```diff
@@ -108,3 +108,25 @@ export type Infer<T> = T extends DmlEntity<infer Schema>
         : Schema[K]["$dataType"]
     }>
   : never
+
+/**
+ * Extracts names of relationships from a schema
+ */
+export type ExtractEntityRelations<
+  Schema extends Record<string, any>,
+  OfType extends RelationshipTypes
+> = {
+  [K in keyof Schema & string]: Schema[K] extends RelationshipType<any>
```

Comment:
```suggestion
  [K in keyof Schema]: Schema[K] extends RelationshipType<any>
```
K is already typed as string in record, it should not be necessary

Reply:
Just tried this in TypeScript playground and seems to expand to `string | number | symbol` when a generic is passed to the type. 

https://www.typescriptlang.org/play/?ts=5.1.6#code/C4TwDgpgBAysBOBLAdgcxgYwBYQLYEMAeAFSggA9gJkATAZygCUIMB7eGwuhFVAGijIArrgBGEeAD5JUALxQA3gCgoUANoBpKCigBrCCFYAzKMQC6ALlOazZStXqCR4+FAD8ULVeQQAbhKUAXzV9QxNzNTMlJSMhZAxgRFZkPQM6QkwcAjsqWgZmNg4uHjQBYTEJSQEtClzHOCQ0TLwiZoJpAApdKw0ASkUVPQA6AjAOjsR+2RllVVVEIeBWABlWAHcJAGF8OggO3sHAg8CgA